### ISSUE TITLE
non-abstract subtypes inherit abstract attributes, but throw exception in the following define statement

### ISSUE THREAD

**Issue by JonThom** [vaticle/typedb#6516]
## Description

When defining (1) an abstract attribute `abstract_attribute` (2) an abstract type `abstract_entity` (3) a non-abstract subtype of abstract_entity, `concrete_entity` 
and committing, 
and then defining that `abstract_entity owns abstract_attribute` 
we see no exception.

However, an exception occurs next time we make a define statement.

## Environment

1. OS (where TypeDB server runs): MacS 12.1
2. TypeDB version (and platform): TypeDB 2.6.0
3. TypeDB client: client-python, console }

## Reproducible Steps

given the following schema 

```
define
    abstract_attribute sub attribute,
        value string,
        abstract;

    abstract_entity sub entity,
        abstract;
        
    concrete_entity sub abstract_entity; 
```
commit

if we make abstract_entity own abstract_attribute

```
define abstract_entity owns abstract_attribute;

```
commit

we don't get an exception - which is surprising, as the ownership of abstract_attribute will be  inherited by concrete_entity, which is a no-no.

and, slightly confusingly, next time we include concrete_entity in a define statement, we indeed do get an exception referring back to that prior transgression
```
define myrel sub relation,
    relates myrole;

  define concrete_entity plays myrel:myrole;
```
commit

## Expected Output

Excepted an exception at `define abstract_entity owns abstract_attribute;`

## Actual Output

> [TYW08] Invalid Type Write: The type 'concrete_entity' is not abstract, and thus cannot own an abstract attribute type 'abstract_attribute'.

**Comment by jmsfltchr** [comment]
Very intriguing! Thanks for reporting. @flyingsilverfin is the best person for this

**Comment by dmitrii-ubskii** [comment]
Fixed in 2.13:
```
> transaction abstract schema write
abstract::schema::write> define
                        abstract_attribute sub attribute, value string, abstract;
                        abstract_entity sub entity, abstract;
                        concrete_entity sub abstract_entity; 

Concepts have been defined
abstract::schema::write*> commit
Transaction changes committed

> transaction abstract schema write
abstract::schema::write> define abstract_entity owns abstract_attribute;

Concepts have been defined
abstract::schema::write*> commit
[TYW08] Invalid Type Write: The type 'concrete_entity' is not abstract, and thus cannot own an abstract attribute type 'abstract_attribute'.
[TYW08] Invalid Type Write: The type 'concrete_entity' is not abstract, and thus cannot own an abstract attribute type 'abstract_attribute'.
```
